### DISCUSSION TITLE
How to run with a specific backend?

### DISCUSSION THREAD

**Discussion by AngryPenguinPL** [ggml-org/llama.cpp#10898] [General]
I wonder if there is an option to run llama.cpp with a specific backend?

That is, I have llama compiled in a combined cpu as well as vulkan and opencl.
Can I add some variable to the cli to force the use of a specific backend?

**Comment by ericcurtin** [comment]
I would like if the various bankends could be dlopen'd, would make packaging easier for Linux distros. One could install additional libraries for other backends if they feel the need to do so.

**Comment by slaren** [comment]
@AngryPenguinPL please take a look at #10497
@ericcurtin this was added in #10469

**Reply by ericcurtin** [reply]
Oh yes sorry, cool :)
